chips/lowrisc/src/registers/adc_ctrl_regs.rs

```diff
@@ -0,0 +1,129 @@
+// Generated register constants for adc_ctrl
+
+// Built for earlgrey_silver_release_v5-5548-g933e261c9
+// https://github.com/lowRISC/opentitan/tree/933e261c91e67309b242d87a97e42b8af1bd17ef
+// Tree status: modified
+// Build date: 2022-05-24T16:20:07
+
+// Copyright information found in source file:
+// Copyright lowRISC contributors.
+
+// Licensing information found in source file:
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
This should also probably be updated to match Tock's LICENSE

Reply:
Can you clarify what you mean?  AFAICT, tock is dual-licensed under the Apache 2.0 and MIT licenses.  This comment identifies the license as an Apache 2.0 license.

Reply:
*Disclaimer: not a lawyer*. Doesn't being dual-licensed under the Apache-2.0 and MIT license mean that the entire codebase can be used under the terms and conditions of *either* of those licenses? This would imply that all contributions to Tock must grant usage and distribution rights up to the extend defined by *both* the Apache-2.0 and MIT license. If I understand this correctly, if only a single file were not dual-licensed, effectively the codebase in its entire would no longer be dual-licensed. In that case, we are not able to accept any file with an explicit license header, except if it declares an Apache-2.0/MIT dual licensing scheme as defined for the rest of Tock.

Reply:
I mean it should be dual licensed as described here: https://github.com/tock/tock#license

But also there is no `LICENSE` file for details in Tock, so this should point to [LICENSE-APACHE](https://github.com/tock/tock/blob/master/LICENSE-APACHE) and [LICENSE-MIT](https://github.com/tock/tock/blob/master/LICENSE-MIT).

It's also a little odd that it is a different copyright then all the other files, but I don't think that's a blocking issue.

Reply:
The code in this PR is dual-licensed under the Apache 2.0 and MIT licenses, as is required by the Tock project. There is no license header because the Tock project has elected not to use license headers.

I'll reiterate that OpenTitan would prefer for Tock to adopt license headers. I suspect that having license headers would have avoided this conversation.

Reply:
@jrvanwhy I am a little confused, do you disagree that one of the following needs to happen:

a) remove this license header, since it implies that this file is only apache licensed and points to a LICENSE file that does not exist in this repository OR

b) update this header as per https://github.com/lowRISC/opentitan/pull/12963 such that it clearly specifies both Apache and MIT licenses

Reply:
> @jrvanwhy I am a little confused, do you disagree that one of the following needs to happen:
>
> a) remove this license header, since it implies that this file is only apache licensed and points to a LICENSE file that does not exist in this repository OR

This isn't a license header. These files were generated by a tool, and that tool mentions the license of some of its input files in its output, but that license is not the license of the output files.
 
> b) update this header as per [lowRISC/opentitan#12963](https://github.com/lowRISC/opentitan/pull/12963) such that it clearly specifies both Apache and MIT licenses

That's not necessary, but very possible.

If I recall correctly, the last time we discussed license headers in Tock, there was consensus that we should either have license headers on all files or license headers on no files.

Reply:
> This isn't a license header. These files were generated by a tool, and that tool mentions the license of some of its input files in its output, but that license is not the license of the output files.

Okay, I actually missed this part. I think it's totally fine to include the licese of source files, but (a) I don't fully understand why we'd want to do this and (b) we should definitely add some additional markup to make it immediately obvious this isn't trying to apply a license to this file as part of the Tock codebase. Perhaps add some indentation?

Reply:
I've updated the generated files such that they (a) explicitly state they're licensed under either Apache or MIT, and (b) explicitly state the original source location and licence of the reference definitions.

Reply:
I am happy with the updated text

Reply:
I'm worried the incorrect `SPDX-License-Identifier` will break automation tools. Can we just remove the original source license information?

Reply:
I'm a big fan of license headers...

Reply:
Great! But this isn't the correct license header. Anyone or any tool looking at the `SPDX-License-Identifier` will see the incorrect license